src/Microsoft.AspNet.Mvc.ViewFeatures/ViewFeatures/DefaultHtmlGenerator.cs

```diff
@@ -261,8 +270,8 @@ public virtual IHtmlContent GenerateAntiforgery(ViewContext viewContext)
                 throw new ArgumentNullException(nameof(viewContext));
             }
 
-            var action =
-                _urlHelper.RouteUrl(routeName, values: routeValues, protocol: null, host: null, fragment: null);
+            var urlHelper = _urlHelperFactory.Create(viewContext);
```

Comment:
> we typically will have 1 `DefaultHtmlGenerator` per view context.

Huh? This class is registered as scoped in DI and, with @rynowak's changes, can become a singleton.